src/zc/buildout/patches.py

```diff
@@ -56,3 +56,113 @@ def set_location(self, l):
 
 
 patch_Distribution()
+
+
+def patch_PackageIndex():
+    try:
+        from setuptools.package_index import PackageIndex
+        from setuptools.package_index import URL_SCHEME
+        from setuptools.package_index import HREF
+        from setuptools.package_index import distros_for_url
+        from setuptools.package_index import htmldecode
+
+        from pip._internal.index.collector import HTMLPage
+        from pip._internal.index.collector import parse_links
+        from pip._internal.index.collector import _clean_link
+        from pip._internal.index.package_finder import _check_link_requires_python
+        from pip._internal.models.target_python import TargetPython
+        from pip._vendor import six
+        from pip._vendor.six.moves import urllib
+
+        PY_VERSION_INFO = TargetPython().py_version_info
+
+        def process_url(self, url, retrieve=False):
```

Comment:
Might be a good idea to comment that this method was copied and pasted from setuptools.package_index.PackageIndex.process_url from setuptools version XX.YY, and then add box comments around the places where it was modified.

Reply:
What do you mean by box comments ?

Reply:
```python
... upstream code ...
# --- LOCAL CHANGES MADE HERE: ---
... modified code ...
# --- END OF LOCAL CHANGES ---
... upstream code ...
```